modules/navier_stokes/doc/content/source/problems/NavierStokesProblem.md

```diff
@@ -88,6 +88,66 @@ A final option available to users is the [!param](/Problem/NavierStokesProblem/u
+preconditioners for "high" Reynolds numbers. The overarching result from that
+paper is that Olshanskii is better performing, although the largest kinematic
+viscosity explored in that article direclty comparing the two methods is
+$10^{-1}$. The Olshanskii preconditioner is known to be sub-optimal for high
+viscosities. In the limit of Stokes flow in which the advective term is dropped,
+the Schur complement preconditioned with Olshanskii has a condition number that
+is the square of the optimally preconditioned (using the pressure mass matrix)
+Schur complement, which leads roughly to a doubling in iterations for the Schur
+complement solve (see [!citep](olshanskii2007pressure)). Below we compare three
+different preconditioning methods for the Schur complement, Elman LSC,
+Olshanskii LSC, and direct preconditioning of the Schur complement with the
+pressure mass matrix, for three different kinematic viscosities and mesh sizes
+for the lid driven cavity with a peak lid velocity of 1 and mesh dimension 1
+(consequently the Reynolds number is the reciprocal of the kinematic
+viscosity). For this test, the Elman preconditioner outperforms Olshanskii at
+high viscosities. However, for low viscosities, Elman shows an iteration count
+that scales with the mesh size whereas Olshanskii iteration counts are
+independent of the mesh size. This is likely due to the better clustering of
+eigenvalues by the Olshanskii preconditioner compared to Elman shown in
+[!citep](zanetti2020scalable). Direct preconditioning with the pressure mass
+matrix is very effective for high viscosities (as theory predicts), but performs
+worse than Olshanskii (and Elman for smaller meshes) for low viscosities. In
+summary, we recommend that users use Elman or the pressure mass matrix directly
+for high viscosities (low Reynolds numbers) and Olshanskii for low viscosities
+(high Reynolds numbers)
+
+Elman:
+
+| $nu$      | n = 20 | n = 40 | n = 80 |
+| ---       | ---    | ---    | ---    |
+| 100       | 5      | 6      | 8      |
+| 1         | 5      | 7      | 8      |
+| $10^{-2}$ | 13     | 16     | 22     |
+
+Olshanskii:
+
+| $nu$      | n = 20 | n = 40 | n = 80 |
+| ---       | ---    | ---    | ---    |
+| 100       | 11     | 12     | 12     |
+| 1         | 11     | 12     | 12     |
+| $10^{-2}$ | 17     | 18     | 17     |
+
+Pressure
+
+| $nu$      | n = 20 | n = 40 | n = 80 |
+| ---       | ---    | ---    | ---    |
+| 100       | 6      | 6      | 6      |
+| 1         | 6      | 6      | 6      |
+| $10^{-2}$ | 21     | 21     | 21     |
+
+The tables above were created by running the `steady_vector_fsp_elman.i`,
+`steady_vector_fsp.i`, and `steady_vector_fsp_stokes.i` inputs respectively. To
+focus the comparison on the preconditioning of the Schur complement itself, all
+multigrid preconditioners were replaced with LU decompositions (`-pc_type lu
+-pc_factor_mat_solver_type mumps`). In the `steady_vector_fsp_stokes.i` input,
```

Comment:
I want to see the result of "-snes_view" where I will have every single detail of your setup.

Reply:
<details>
<summary>Example SNES view</summary>
```
SNES Object: 1 MPI process
  type: newtonls
  maximum iterations=50, maximum function evaluations=10000
  tolerances: relative=1e-08, absolute=1e-50, solution=0.
  total number of linear solver iterations=8
  total number of function evaluations=4
  norm schedule ALWAYS
  SNESLineSearch Object: 1 MPI process
    type: bt
      interpolation: cubic
      alpha=1.000000e-04
    maxstep=1.000000e+08, minlambda=1.000000e-12
    tolerances: relative=1.000000e-08, absolute=1.000000e-15, lambda=1.000000e-08
    maximum iterations=40
  KSP Object: 1 MPI process
    type: fgmres
      restart=300, using Classical (unmodified) Gram-Schmidt Orthogonalization with no iterative refinement
      happy breakdown tolerance 1e-30
    maximum iterations=10000, initial guess is zero
    tolerances:  relative=0.0001, absolute=1e-50, divergence=1e+100
    right preconditioning
    using UNPRECONDITIONED norm type for convergence test
  PC Object: 1 MPI process
    type: fieldsplit
      FieldSplit with Schur preconditioner, factorization FULL
      Preconditioner for the Schur complement formed from S itself
      Split info:
      Split number 0 Defined by IS
      Split number 1 Defined by IS
      KSP solver for A00 block
        KSP Object: (fieldsplit_u_) 1 MPI process
          type: gmres
            restart=300, using Classical (unmodified) Gram-Schmidt Orthogonalization with no iterative refinement
            happy breakdown tolerance 1e-30
          maximum iterations=10000, initial guess is zero
          tolerances:  relative=0.01, absolute=1e-50, divergence=10000.
          right preconditioning
          using UNPRECONDITIONED norm type for convergence test
        PC Object: (fieldsplit_u_) 1 MPI process
          type: hypre
            HYPRE BoomerAMG preconditioning
              Cycle type V
              Maximum number of levels 25
              Maximum number of iterations PER hypre call 1
              Convergence tolerance PER hypre call 0.
              Threshold for strong coupling 0.25
              Interpolation truncation factor 0.
              Interpolation: max elements per row 0
              Number of levels of aggressive coarsening 0
              Number of paths for aggressive coarsening 1
              Maximum row sums 0.9
              Sweeps down         1
              Sweeps up           1
              Sweeps on coarse    1
              Relax down          symmetric-SOR/Jacobi
              Relax up            symmetric-SOR/Jacobi
              Relax on coarse     Gaussian-elimination
              Relax weight  (all)      1.
              Outer relax weight (all) 1.
              Maximum size of coarsest grid 9
              Minimum size of coarsest grid 1
              Using CF-relaxation
              Not using more complex smoothers.
              Measure type        local
              Coarsen type        Falgout
              Interpolation type  classical
              SpGEMM type         hypre
          linear system matrix = precond matrix:
          Mat Object: (fieldsplit_u_) 1 MPI process
            type: seqaij
            rows=578, cols=578
            total: nonzeros=16900, allocated nonzeros=16900
            total number of mallocs used during MatSetValues calls=0
              using I-node routines: found 266 nodes, limit used is 5
      KSP solver for S = A11 - A10 inv(A00) A01
        KSP Object: (fieldsplit_p_) 1 MPI process
          type: fgmres
            restart=300, using Classical (unmodified) Gram-Schmidt Orthogonalization with no iterative refinement
            happy breakdown tolerance 1e-30
          maximum iterations=10000, initial guess is zero
          tolerances:  relative=0.01, absolute=1e-50, divergence=10000.
          right preconditioning
          using UNPRECONDITIONED norm type for convergence test
        PC Object: (fieldsplit_p_) 1 MPI process
          type: lsc
            KSP Object: (fieldsplit_p_lsc_) 1 MPI process
              type: fgmres
                restart=300, using Classical (unmodified) Gram-Schmidt Orthogonalization with no iterative refinement
                happy breakdown tolerance 1e-30
              maximum iterations=10000, initial guess is zero
              tolerances:  relative=0.1, absolute=1e-50, divergence=10000.
              right preconditioning
              using UNPRECONDITIONED norm type for convergence test
            PC Object: (fieldsplit_p_lsc_) 1 MPI process
              type: hypre
                HYPRE BoomerAMG preconditioning
                  Cycle type V
                  Maximum number of levels 25
                  Maximum number of iterations PER hypre call 1
                  Convergence tolerance PER hypre call 0.
                  Threshold for strong coupling 0.25
                  Interpolation truncation factor 0.
                  Interpolation: max elements per row 0
                  Number of levels of aggressive coarsening 0
                  Number of paths for aggressive coarsening 1
                  Maximum row sums 0.9
                  Sweeps down         1
                  Sweeps up           1
                  Sweeps on coarse    1
                  Relax down          symmetric-SOR/Jacobi
                  Relax up            symmetric-SOR/Jacobi
                  Relax on coarse     Gaussian-elimination
                  Relax weight  (all)      1.
                  Outer relax weight (all) 1.
                  Maximum size of coarsest grid 9
                  Minimum size of coarsest grid 1
                  Using CF-relaxation
                  Not using more complex smoothers.
                  Measure type        local
                  Coarsen type        Falgout
                  Interpolation type  classical
                  SpGEMM type         hypre
              linear system matrix = precond matrix:
              Mat Object: 1 MPI process
                type: seqaij
                rows=578, cols=578
                total: nonzeros=16900, allocated nonzeros=16900
                total number of mallocs used during MatSetValues calls=0
                  using I-node routines: found 266 nodes, limit used is 5
            KSP Object: (fieldsplit_p_lsc_mass_) 1 MPI process
              type: gmres
                restart=30, using Classical (unmodified) Gram-Schmidt Orthogonalization with no iterative refinement
                happy breakdown tolerance 1e-30
              maximum iterations=10000, initial guess is zero
              tolerances:  relative=0.1, absolute=1e-50, divergence=10000.
              left preconditioning
              using PRECONDITIONED norm type for convergence test
            PC Object: (fieldsplit_p_lsc_mass_) 1 MPI process
              type: hypre
                HYPRE BoomerAMG preconditioning
                  Cycle type V
                  Maximum number of levels 25
                  Maximum number of iterations PER hypre call 1
                  Convergence tolerance PER hypre call 0.
                  Threshold for strong coupling 0.25
                  Interpolation truncation factor 0.
                  Interpolation: max elements per row 0
                  Number of levels of aggressive coarsening 0
                  Number of paths for aggressive coarsening 1
                  Maximum row sums 0.9
                  Sweeps down         1
                  Sweeps up           1
                  Sweeps on coarse    1
                  Relax down          symmetric-SOR/Jacobi
                  Relax up            symmetric-SOR/Jacobi
                  Relax on coarse     Gaussian-elimination
                  Relax weight  (all)      1.
                  Outer relax weight (all) 1.
                  Maximum size of coarsest grid 9
                  Minimum size of coarsest grid 1
                  Using CF-relaxation
                  Not using more complex smoothers.
                  Measure type        local
                  Coarsen type        Falgout
                  Interpolation type  classical
                  SpGEMM type         hypre
              linear system matrix = precond matrix:
              Mat Object: 1 MPI process
                type: seqaij
                rows=81, cols=81
                total: nonzeros=625, allocated nonzeros=625
                total number of mallocs used during MatSetValues calls=0
                  not using I-node routines
          linear system matrix = precond matrix:
          Mat Object: (fieldsplit_p_) 1 MPI process
            type: schurcomplement
            rows=81, cols=81
              Schur complement A11 - A10 inv(A00) A01
              A11
                Mat Object: (fieldsplit_p_) 1 MPI process
                  type: seqaij
                  rows=81, cols=81
                  total: nonzeros=81, allocated nonzeros=81
                  total number of mallocs used during MatSetValues calls=0
                    not using I-node routines
              A10
                Mat Object: 1 MPI process
                  type: seqaij
                  rows=81, cols=578
                  total: nonzeros=3362, allocated nonzeros=3362
                  total number of mallocs used during MatSetValues calls=0
                    not using I-node routines
              KSP solver for A00 block viewable with the additional option -fieldsplit_u_ksp_view
              A01
                Mat Object: 1 MPI process
                  type: seqaij
                  rows=578, cols=81
                  total: nonzeros=3362, allocated nonzeros=3362
                  total number of mallocs used during MatSetValues calls=0
                    using I-node routines: found 266 nodes, limit used is 5
    linear system matrix = precond matrix:
    Mat Object: () 1 MPI process
      type: seqaij
      rows=659, cols=659
      total: nonzeros=23705, allocated nonzeros=24249
      total number of mallocs used during MatSetValues calls=0
        using I-node routines: found 347 nodes, limit used is 5
```
</details>